qt/preferences_dialog.cpp

```diff
@@ -97,6 +99,28 @@ namespace qt
       });
     }
 
+    QLabel * mapLanguageLabel = new QLabel("Map Language");
+    QComboBox * mapLanguageComboBox = new QComboBox();
+    {
+      StringUtf8Multilang::Languages const & supportedLanguages = StringUtf8Multilang::GetSupportedLanguages();
+      QStringList languagesList = QStringList();
+      for (auto const & language : supportedLanguages)
+        languagesList << QString::fromStdString(std::string(language.m_name));
+
+      mapLanguageComboBox->addItems(languagesList);
+      mapLanguageComboBox->setCurrentIndex(StringUtf8Multilang::GetLangIndex(languages::GetCurrentMapLanguageCode()));
+      connect(mapLanguageComboBox, &QComboBox::currentIndexChanged, [&framework](int index)
+      {
+        auto const & mapLanguageCode = std::string(StringUtf8Multilang::GetLangByCode(index));
+        if (mapLanguageCode.empty()) // Empty means unsupported
```

Comment:
Aren't all languages retrieved by `StringUtf8Multilang::GetSupportedLanguages()` supported? Isn't this check redundant and may be removed or replaced by the ASSERT?